sympy/matrices/immutable.py

```diff
@@ -101,18 +98,6 @@ def _eval_extract(self, rowsList, colsList):
         return self._new(len(rowsList), len(colsList),
                          Tuple(*(mat[i] for i in indices), sympify=False), copy=False)
 
-    @property
-    def cols(self):
-        return int(self.args[1])
-
-    @property
-    def rows(self):
-        return int(self.args[0])
-
-    @property
-    def shape(self):
```

Comment:
It doesn't look like it has the shape attribute.

Reply:
Shape attribute is defined in matrix common and it is accessible after creation
`ImmutableDenseMatrix([[1, 2], [3, 4]]).shape`